modules/integration/tests-ui-integration/src/test/java/org/wso2/carbon/mb/ui/test/dlc/DLCDurableTopicTestCase.java

```diff
@@ -0,0 +1,284 @@
+        AndesClientUtils.sleepForInterval(80000L);
+
+    }
+
+
+    /**
+     * This test will verify delete and restore functions in dlc for durable topic messages.
+     * 1. delete first element in dlc table. if message id doesn't exist in dlc ui table after
+     * delete test will be success.
+     * 2. restore first element in dlc table. if consumer client receive new messages after restore
+     * from dlc test will be success.
+     * TODO: After reroute feature completed in ui there should be a test for reroute durable topic messages.
+     *
+     * @throws Exception
+     */
+    @Test()
+    public void performDurableTopicDeadLetterChannelTestCase() throws Exception {
+
+        String deletingMessageID;
+        String restoringMessageID;
+
+        DLCBrowsePage dlcBrowsePage = homePage.getDLCBrowsePage();
+        Assert.assertNotNull(dlcBrowsePage.isDLCCreated(),
+                             "DeadLetter Channel not created. " + DLC_TEST_DURABLE_TOPIC);
+        //Testing delete messages
+        DLCContentPage dlcContentPage = dlcBrowsePage.getDLCContent();
+        deletingMessageID = dlcContentPage.deleteFunction();
+
+        Assert.assertTrue(checkMessages(deletingMessageID, DLC_TEST_DURABLE_TOPIC),
+                          "Deleting messages of dead letter channel is unsuccessful.");
+
+        // number of messages received by consumer client before restore function triggered.
+        Long beforeRestoreMessageReceivedCount = consumerClient.getReceivedMessageCount();
+
+        //Testing restore messages
+        restoringMessageID = dlcContentPage.restoreFunction();
+        waitUntilMessagesReceived(9L);
+
+        //Thread sleep until messages sent to DLC after breaching maximum number of retrying
+        AndesClientUtils.sleepForInterval(80000L);
+
+        // number of messages received by consumer client after restore function triggered.
+        Long afterRestoreMessageReceivedCount = consumerClient.getReceivedMessageCount();
+
+        // This assertion will check if consumer client has received messages messages after
+        // restore function triggered from ui. If it receives messages after restore function triggered
+        // this assertion will be success.
+        Assert.assertTrue(beforeRestoreMessageReceivedCount < afterRestoreMessageReceivedCount,
+                          restoringMessageID + "Durable topic message not successfully restored.");
+    }
+
+    /**
+     * This method will wait until given number of messages received by the consumer client.
+     *
+     * @param numberOfMessages
+     */
+    public void waitUntilMessagesReceived(Long numberOfMessages) {
+
+        boolean isRetryInProgress = true;
+
+        while (isRetryInProgress) {
+            Long count = consumerClient.getReceivedMessageCount();
+            if (count >= numberOfMessages) {
+                isRetryInProgress = false;
```

Comment:
Would it make sense to have a while(true) loop and have a break inside the if condition ?